### PR TITLE
Package: All Versions / Recent Version buttons

### PR DESCRIPTION
**Pull request by eksperimental**
It changes the previous behaviour to the text is toggled on click.

Closes: https://github.com/hexpm/hex_web/issues/143

### CODE REVIEW

File: lib/hex_web/web/templates/main.html.eex [hunk 1]
```diff
@@ -73,6 +73,7 @@
 
     <script src="//cdnjs.cloudflare.com/ajax/libs/jquery/2.1.3/jquery.min.js"></script>
     <script src="//cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.1/js/bootstrap.min.js"></script>
+    <script src="/static/main.js?<%= asset_id %>"></script>
```
Comment (ericmj) [inline lib/hex_web/web/templates/main.html.eex]: What is this for?

Comment (eksperimental) [inline lib/hex_web/web/templates/main.html.eex]: sorry that file was not staged,
I have updated the commit, let me know if you want the code inside the template.

**Comment by ericmj** [comment]
Can we keep it uncollapsed for people without javascript?

**Comment by eksperimental** [comment]
sure! but then I think I need to refactor the whole thing, as it will be better to have one list, and just toggle whatever elements we don't want to show if javascript is available.

**Comment by ericmj** [comment]
Ping? :)

**Comment by eksperimental** [comment]
sorry about the delay @ericmj , I will get my hand on this tomorrow (I am finishing something for Elixir 1.1.0)

**Comment by eksperimental** [comment]
done :-)

File: priv/static/static/main.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// Hide older versions on load
+$('#versions li.older').collapse('hide');
+
+// Package: toggle text in "All Versions / Recent Version" buttons
+$('.show-versions .invisible').removeClass('invisible').toggle();
+$('.show-versions .toggle-text').click(function() {
+  $(this).find('span').each(function() {
+    $(this).toggle();
```
Comment (henrik) [inline priv/static/static/main.js]: I think `$(this).find('span').toggle();` would also work here.

**Comment by ericmj** [comment]
Thank you @eksperimental. Will test this locally and then merge.

File: priv/static/static/main.js (hunk 2 above)
Comment (eksperimental) [inline priv/static/static/main.js]: thanks @henrik, I have updated the code.

**Comment by eksperimental** [comment]
I have updated the code and rebased it with the current master,
including suggestion by @henrik and also by default the show all versions button is disabled, only shown when JS is enabled.

@ericmj the build is failing with elixir master on OTP 17.x,
locally runnning elixir 1.1.0-beta on OTP 18 pass all tests including integration

Could be a non-backward compatible changed recently introduced in elixir master when running OTP 17?

**Comment by ericmj** [comment]
Thank you. Will test this locally.

**Comment by ericmj** [comment]
Looked great. Thanks!

**Pull request merged**